fl4health/utils/load_data.py

```diff
@@ -1,85 +1,138 @@
+        training_set = dataset_converter.convert_dataset(training_set)
+        validation_set = dataset_converter.convert_dataset(validation_set)
 
-    train_loader = DataLoader(train_ds, batch_size=batch_size, shuffle=True)
-    validation_loader = DataLoader(val_ds, batch_size=batch_size)
+    train_loader = DataLoader(training_set, batch_size=batch_size, shuffle=True)
+    validation_loader = DataLoader(validation_set, batch_size=batch_size)
 
-    num_examples = {"train_set": len(train_ds), "validation_set": len(val_ds)}
+    num_examples = {"train_set": len(training_set), "validation_set": len(validation_set)}
     return train_loader, validation_loader, num_examples
 
 
-def load_cifar10_test_data(
-    data_dir: Path, batch_size: int, sampler: Optional[LabelBasedSampler] = None
-) -> Tuple[DataLoader, Dict[str, int]]:
-    """Load CIFAR-10 test set only."""
-    log(INFO, f"Data directory: {str(data_dir)}")
-    transform = transforms.Compose(
-        [
-            transforms.ToTensor(),
-            transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)),
-        ]
-    )
-    evaluation_set: BaseDataset = Cifar10Dataset(data_dir, train=False, transform=transform)
+def get_cifar10_data_and_target_tensors(data_dir: Path, train: bool) -> Tuple[torch.Tensor, torch.Tensor]:
+    mnist_dataset = CIFAR10(data_dir, train=train, download=True)
+    data = torch.Tensor(mnist_dataset.data)
+    targets = torch.Tensor(mnist_dataset.targets).long()
+    return data, targets
 
-    if sampler is not None:
-        evaluation_set = sampler.subsample(evaluation_set)
 
-    evaluation_loader = DataLoader(evaluation_set, batch_size=batch_size, shuffle=False)
-    num_examples = {"eval_set": len(evaluation_set)}
-    return evaluation_loader, num_examples
+def get_train_and_val_cifar10_datasets(
+    data_dir: Path,
+    transform: Optional[Callable] = None,
+    target_transform: Optional[Callable] = None,
+    validation_proportion: float = 0.2,
+) -> Tuple[TensorDataset, TensorDataset]:
+    data, targets = get_cifar10_data_and_target_tensors(data_dir, True)
+
+    train_data, train_targets, val_data, val_targets = split_data_and_targets(data, targets, validation_proportion)
+
+    training_set = TensorDataset(train_data, train_targets, transform=transform, target_transform=target_transform)
+    validation_set = TensorDataset(val_data, val_targets, transform=transform, target_transform=target_transform)
+
+    return training_set, validation_set
 
 
 def load_cifar10_data(
-    data_dir: Path, batch_size: int, sampler: Optional[LabelBasedSampler] = None
+    data_dir: Path,
+    batch_size: int,
+    sampler: Optional[LabelBasedSampler] = None,
+    validation_proportion: float = 0.2,
+    seed: int = 42,
 ) -> Tuple[DataLoader, DataLoader, Dict[str, int]]:
     """Load CIFAR-10 (training and validation set)."""
     log(INFO, f"Data directory: {str(data_dir)}")
+
+    set_all_random_seeds(seed)
```

Comment:
Same comment about the seeds here. That is, we need to only do it locally and have the provided seed be optional and none by default.